### PR TITLE
subjects: label facets

### PR DESCRIPTION
**Pull request by fenekku**
- closes https://github.com/inveniosoftware/invenio-vocabularies/issues/68
- release: v0.8.1

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -31,6 +31,9 @@ var/
 *.egg-info/
 .installed.cfg
 *.egg
+Pipfile
+Pipfile.lock
+pyproject.toml
```
Comment (ppanero) [inline .gitignore:36]: If we are gonna add this to every module we should add them in the `cookiecutter-invenio-module` so they are there from the initial commit.

File: invenio_vocabularies/contrib/subjects/facets.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+"""Subjects services."""
+
+
+class SubjectsLabels:
+    """Fetching of subjects labels for facets."""
+
+    def __call__(self, ids):
+        """Return the mapping when evaluated.
+
+        For subjects, the ids received are actually human-readable and coming
```
Comment (ppanero) [inline invenio_vocabularies/contrib/subjects/facets.py:18]: Is this the case for every subject scheme items? I understand the scheme itself is ok (e.g. MeSH is a human readable) but the items... i.e.

- MeSH: https://raw.githubusercontent.com/galterlibrary/invenio-subjects-mesh/master/invenio_subjects_mesh/vocabularies/subjects_mesh.yaml
- FOS: https://github.com/inveniosoftware/cookiecutter-invenio-rdm/pull/189/files

They both have a URL in the `id` field. I'm wondering then what's the frontend sending as id if it's human readable 🤔

**Review by ppanero — COMMENTED** [review]
I might be missing some context. So if blocking go ahead. I am wondering about the id *as-is* in the labels.

File: tests/contrib/subjects/test_subjects_facets.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+
+def test_subjects_label_duplicates():
+    labels_instance = SubjectsLabels()
+
+    result = labels_instance(["MeSH", "MeSH"])
+
+    assert {"MeSH": "MeSH"} == result
+
+    result = labels_instance(["Abdomen", "Abdomen", "Abdomen"])
```
Comment (ppanero) [inline tests/contrib/subjects/test_subjects_facets.py:21]: Wouldn't we receive the URL here? I'm not sure

File: .gitignore (hunk 1 above)
Comment (fenekku) [reply .gitignore:36]: Sure... created here: https://github.com/inveniosoftware/cookiecutter-invenio-module/pull/145

File: invenio_vocabularies/contrib/subjects/facets.py (hunk 2 above)
Comment (fenekku) [reply invenio_vocabularies/contrib/subjects/facets.py:18]: Ah maybe `ids` is what is tripping up here. These `ids` are actually either the vocabulary `scheme` (top-level) or `subject` (nested level). Not the vocabulary `id`. That's what the comment was about. `ids` as an argument was kept to be *consistent* with the interface of the `Labels` category of classes (e.g. [`VocabularyLabels`](https://github.com/inveniosoftware/invenio-vocabularies/blob/master/invenio_vocabularies/services/facets.py#L39)).

What about `label_keys` as a better argument name for both this class and `VocabularyLabels` ?

File: tests/contrib/subjects/test_subjects_facets.py (hunk 3 above)
Comment (fenekku) [reply tests/contrib/subjects/test_subjects_facets.py:21]: No, see above.

File: invenio_vocabularies/contrib/subjects/facets.py (hunk 2 above)
Comment (ppanero) [reply invenio_vocabularies/contrib/subjects/facets.py:18]: Ah I see, thanks. Maybe clarifying it in the docstring is enough (like putting there almost your answer from above).

Comment (fenekku) [reply invenio_vocabularies/contrib/subjects/facets.py]: ok I just reworded the existing comment.

**Comment by fenekku** [comment]
Ok merging this one from talk in standup

**Pull request merged**